app/actors/hyrax/actors/file_set_actor.rb

```diff
@@ -10,6 +10,40 @@ def initialize(file_set, user)
         @user = user
       end
 
+      # @!group Asynchronous Operations
+
+      # Spawns asynchronous IngestJob
+      # Called from FileSetsController, AttachFilesToWorkJob, ImportURLJob, IngestLocalFileJob
+      # @param [Hyrax::UploadedFile, File, ActionDigest::HTTP::UploadedFile, Tempfile] file the file uploaded by the user
```

Comment:
You are right that `IngestJob` won't work on a `Tempfile`.  But unlike its predecessor, this method doesn't attempt to pass the `Tempfile` directly to the job.  Instead it calls `IngestJob.perform_later(wrapper!(file, relation))`, where `wrapper!` effectively converts `file` to a persisted `JobIoWrapper` to match the signature of `IngestJob`.  Make sense?